src/formState.ts

```diff
@@ -346,13 +347,16 @@ function newObjectState<T, P = any>(
     },
 
     get readOnly(): boolean {
-      return this._readOnly || this._considerReadOnly();
+      return (
+        this._readOnly ||
+        this._considerReadOnly() ||
+        (parentState && parentState().readOnly) ||
+        parentListState?.readOnly
+      );
```

Comment:
Kind of surprised this will return a boolean... won't this be `boolean | undefined`? I'd think we'd need to wrap this with `Boolean(...)`?

Reply:
Yeah, I was surprised too; fwiw I think it's one of TypeScript's "it's just more pragmatic given how ubiquitous the pattern is" decisions, and so preventing this requires eslint rules like `strict-boolean-expressions`:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md

Which, ironically I just tried setting that in `graphql-service` last week while on a separate lint-related tangent, and damn it's surprising how often we use the "not _technically_ a boolean" idioms, such that I think enabling it would hurt dev ergonomics more than it helps. Although we'd also probably get used to it.

Reply:
Went ahead and added a `!!` for good measure.

Reply:
Lol, shoot